core/trino-main/src/main/java/io/trino/operator/WorkProcessorSourceOperatorAdapter.java

```diff
@@ -228,6 +232,11 @@ private void updateOperatorStats()
             operatorContext.recordDynamicFilterSplitProcessed(currentDynamicFilterSplitsProcessed - previousDynamicFilterSplitsProcessed);
             previousDynamicFilterSplitsProcessed = currentDynamicFilterSplitsProcessed;
         }
+
+        if (currentMetrics != previousMetrics) {
+            operatorContext.setLatestMetrics(currentMetrics);
+            previousMetrics = currentMetrics;
```

Comment:
Why not set it unconditionally?

Reply:
It can be done, but I thought to optimize for the case where `currentMetrics == Metrics.EMPTY` (which currently happens for all connectors that don't implement `ConnectorPageSource#getMetrics`).

Reply:
setLatestMetrics just sets a field. There's no processing involved, so I don't think this optimization really optimizes much. We can add it in the future if we observe that being an actual problem.